examples/quantum_teleportation.py

```diff
@@ -34,26 +34,88 @@
 """
 
 import random
+from typing import Tuple
+
 import numpy as np
 import cirq
+from cirq.sim.simulator import (
+    TStepResult,
+    SimulatesIntermediateState,
+    TSimulationTrialResult,
+    TSimulatorState,
+    TActOnArgs,
+)
+
+
+def _print_bloch(vector):
+    print(
+        'x: ',
+        np.around(vector[0], 4),
+        'y: ',
+        np.around(vector[1], 4),
+        'z: ',
+        np.around(vector[2], 4),
+    )
 
 
-def make_quantum_teleportation_circuit(ranX, ranY):
-    circuit = cirq.Circuit()
+def _run(
+    sim: SimulatesIntermediateState[
+        TStepResult, TSimulationTrialResult, TSimulatorState, TActOnArgs
+    ]
+) -> Tuple[TSimulationTrialResult, TStepResult]:
+    # Initialize our qubit state space.
     msg, alice, bob = cirq.LineQubit.range(3)
+    qubits = (msg, alice, bob)
+    args = sim._create_act_on_args(0, qubits)
+
+    # First we create a bell state circuit and simulate it on the qubits.
+    bell_circuit = cirq.Circuit(cirq.H(alice), cirq.CNOT(alice, bob))
+    list(sim._core_iterator(bell_circuit, args, qubits))
```

Comment:
Sure, I can revert this file's changes. Calling `simulate_moment_steps` is incompatible with this approach because it requires two separate circuits that share the same ActOnArgs. `simulate_moment_steps` hides the ActOnArgs from you.

I'll revert this file's changes and we can revisit if we decide to create public APIs for the two new functions.

Reply:
Updated `simulate_moment_steps` to accept an `ActOnArgs`, and used that here. Also made `create_act_on_args` public.